src/Sylius/Bundle/CoreBundle/Taxation/Strategy/TaxCalculationStrategy.php

```diff
@@ -12,19 +12,79 @@
 namespace Sylius\Bundle\CoreBundle\Taxation\Strategy;
 
 use Sylius\Component\Addressing\Model\ZoneInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
 use Sylius\Component\Core\Model\OrderInterface;
-use Sylius\Component\Core\Taxation\Strategy\AbstractTaxCalculationStrategy;
+use Sylius\Component\Core\Taxation\Applicator\OrderTaxesApplicatorInterface;
+use Sylius\Component\Core\Taxation\Strategy\TaxCalculationStrategyInterface;
+use Webmozart\Assert\Assert;
 
 /**
  * @author Mark McKelvie <mark.mckelvie@reiss.com>
  */
-class TaxCalculationStrategy extends AbstractTaxCalculationStrategy
+final class TaxCalculationStrategy implements TaxCalculationStrategyInterface
 {
     /**
+     * @var string
+     */
+    private $type;
+
+    /**
+     * @var OrderTaxesApplicatorInterface[]
+     */
+    private $applicators;
+
+    /**
+     * @param string $type
+     * @param OrderTaxesApplicatorInterface[] $applicators
+     */
+    public function __construct($type, array $applicators)
+    {
+        $this->assertApplicatorsHaveCorrectType($applicators);
+
+        $this->type = $type;
+        $this->applicators = $applicators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function applyTaxes(OrderInterface $order, ZoneInterface $zone)
+    {
+        foreach ($this->applicators as $applicator) {
+            $applicator->apply($order, $zone);
+        }
+    }
+
+    /**
      * {@inheritdoc}
      */
     public function supports(OrderInterface $order, ZoneInterface $zone)
     {
-        return $order->getChannel()->getTaxCalculationStrategy() === $this->type;
+        $channel = $order->getChannel();
+
+        /** @var ChannelInterface $channel */
+        Assert::isInstanceOf($channel, ChannelInterface::class);
```

Comment:
How it is possible that `getChannel` method will return something different then instance of `ChannelInterface`? Maybe `Assert::notNull` would be a better solution.